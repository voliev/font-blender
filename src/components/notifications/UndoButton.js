import React from "react";
import PropTypes from "prop-types";
import "./styles/UndoButton.css";

const UndoButton = ({ onClickHandler }) => (
  <button
    className="UndoButton btn"
    onClick={onClickHandler}
    value="undo"
    title="Undo"
  >
    Undo
  </button>
);

UndoButton.propTypes = {
  onClickHandler: PropTypes.func.isRequired
};

export default UndoButton;
