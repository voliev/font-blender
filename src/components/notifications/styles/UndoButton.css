.UndoButton {
  background-color: transparent;
  font-size: var(--font-xs);
  height: 36px;
  min-width: 64px;
  padding: 0;
  color: var(--orange);
  text-transform: uppercase;
}
