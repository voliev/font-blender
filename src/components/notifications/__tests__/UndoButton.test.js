import React from "react";
import { mount, shallow } from "enzyme";
import UndoButton from "../UndoButton";

const setup = (renderFn = shallow, propsOverrides = {}) => {
  const props = {
    onClickHandler: jest.fn(),
    ...propsOverrides
  };

  return {
    ...props,
    wrapper: renderFn(<UndoButton {...props} />)
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { wrapper, onClickHandler } = setup(mount);

  expect(typeof wrapper.prop("onClickHandler")).toBe("function");
  expect(wrapper.prop("onClickHandler")).toBe(onClickHandler);
});

it("should handle button click", () => {
  const { wrapper, onClickHandler } = setup();

  wrapper.find("button").simulate("click");

  expect(onClickHandler).toHaveBeenCalledTimes(1);
});
