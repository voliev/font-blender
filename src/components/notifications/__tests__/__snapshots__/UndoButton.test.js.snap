// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<button
  className="UndoButton btn"
  onClick={[MockFunction]}
  title="Undo"
  value="undo"
>
  Undo
</button>
`;
